Source/Core/Common/CMakeLists.txt

```diff
@@ -117,6 +117,10 @@ if(${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
   set(LIBS ${LIBS} dl)
 endif()
 
+if(OPROFILE_FOUND)
```

Comment:
Done. The majority of the other libraries still use the old pattern, so if that's considered a useful cleanup it could be done as a separate pr I guess